src/main/java/edu/harvard/iq/dataverse/UserServiceBean.java

```diff
@@ -34,5 +41,371 @@ public AuthenticatedUser save( AuthenticatedUser user ) {
+            
+            //String roles = getUserRolesAsString((Integer) dbResultRow[0]);
+            roleString = "";
+            List<String> roleList = roleLookup.get("@" + (String)userInfo[1]);
+            if ((roleList != null)&&(!roleList.isEmpty())){
+                roleString = roleList.stream().collect(Collectors.joining(", "));
+            }
+            AuthenticatedUser singleUser = createAuthenticatedUserForView(userInfo, roleString, rowNum);            
+            viewObjects.add(singleUser);
+        }
+        
+        
+        return viewObjects;
+    }
+    
+    private AuthenticatedUser createAuthenticatedUserForView (Object[] dbRowValues, String roles, int rowNum){
+        AuthenticatedUser user = new AuthenticatedUser();
+        user.setRowNum(rowNum);
+        user.setId(new Long((Integer)dbRowValues[0]));
+        user.setUserIdentifier((String)dbRowValues[1]);
+        user.setLastName(UserUtil.getStringOrNull(dbRowValues[2]));
+        user.setFirstName(UserUtil.getStringOrNull(dbRowValues[3]));
+        user.setEmail(UserUtil.getStringOrNull(dbRowValues[4]));
+        user.setAffiliation(UserUtil.getStringOrNull(dbRowValues[5]));
+        user.setSuperuser((Boolean)(dbRowValues[6]));
+        user.setPosition(UserUtil.getStringOrNull(dbRowValues[7]));
+        user.setModificationTime(UserUtil.getTimestampOrNull(dbRowValues[8]));
+        user.setAuthProviderId(UserUtil.getStringOrNull(dbRowValues[9]));
+        user.setAuthProviderFactoryAlias(UserUtil.getStringOrNull(dbRowValues[10]));
+        
+        user.setRoles(roles);
+        return user;
+    } 
+    
+    /**
+     * Attempt to retrieve all the user roles in 1 query
+     * Consider putting limits on this -- e.g. no more than 1,000 user identifiers or something similar
+     * 
+     * @param userIdentifierList
+     * @return 
+     */
+    private HashMap<String, List<String>> retrieveRolesForUsers(List<String> userIdentifierList){
+        
+        if ((userIdentifierList==null)||(userIdentifierList.isEmpty())){
+            return null;
+        }
+        
+        // Add '@' to each identifier and delimit the list by ","
+        //
+        String identifierList = userIdentifierList.stream()
+                                     .filter(x -> !Strings.isNullOrEmpty(x))
+                                     .map(x -> "'@" + x + "'")
+                                     .collect(Collectors.joining(", "));
+
+        //System.out.println("identifierList: " + identifierList);
+
+        
+        String qstr = "SELECT distinct a.assigneeidentifier,";
+        qstr += " d.name";
+        qstr += " FROM roleassignment a,";
+        qstr += " dataverserole d";
+        qstr += " WHERE d.id = a.role_id";
+        qstr += " AND a.assigneeidentifier IN (" + identifierList + ")";
+        qstr += " ORDER by a.assigneeidentifier, d.name;";
+
+        //System.out.println("qstr: " + qstr);
+
+        Query nativeQuery = em.createNativeQuery(qstr);
```

Comment:
Can we switch this and other non-type safe queries to a TypedQuery or whatever it's called?